src/models/iceflow/SIA2D/SIA2D_utils.jl

```diff
@@ -279,18 +361,18 @@ function surface_V!(H::Matrix{<:Real}, simulation::SIM, t::R, θ) where {SIM <:
 
     D = if iceflow_model.U_is_provided
         # With a U law we can only compute the surface velocity with an approximation as it would require to integrate the diffusivity wrt H
-        U
+        U.value
     elseif iceflow_model.Y_is_provided
         # With a Y law we can only compute the surface velocity with an approximation as it would require to integrate the diffusivity wrt H
-        n_H = iceflow_model.n_H_is_provided ? iceflow_cache.n_H : n
-        n_∇S = iceflow_model.n_∇S_is_provided ? iceflow_cache.n_∇S : n
-        gravity_term = (ρ * g).^n
-        Γ_no_A = @. 2.0 * gravity_term / (n + 2)
-        (C .* gravity_term .+ Y .* Γ_no_A .* H̄) .* H̄.^n_H .* ∇S.^(n_∇S .- 1)
+        n_H = iceflow_model.n_H_is_provided ? iceflow_cache.n_H : n.value
+        n_∇S = iceflow_model.n_∇S_is_provided ? iceflow_cache.n_∇S : n.value
+        gravity_term = (ρ * g).^n.value
+        Γ_no_A = @. 2.0 * gravity_term / (n.value + 2)
+        (C.value .* gravity_term .+ Y.value .* Γ_no_A .* H̄) .* H̄.^n_H .* ∇S.^(n_∇S .- 1)
```

Comment:
Just so I get all these `.value` fields. This is because all these have become caches?